programs/ziti-edge-tunnel/instance.c

```diff
@@ -527,6 +540,78 @@ void update_mfa_time(char* identifier) {
     }
 }
 
+void set_ip_info(uint32_t dns_ip, uint32_t tun_ip, int bits) {
+    tnl_status.TunIpv4Mask = bits;
+
+    if (tnl_status.TunIpv4) free(tnl_status.TunIpv4);
+    ip_addr_t tun_ip4 = IPADDR4_INIT(tun_ip);
+    tnl_status.TunIpv4 = strdup(ipaddr_ntoa(&tun_ip4));
+
+    if (tnl_status.IpInfo) {
+        free_ip_info(tnl_status.IpInfo);
+        free(tnl_status.IpInfo);
+    }
+    ip_addr_t dns_ip4 = IPADDR4_INIT(dns_ip);
+    tnl_status.IpInfo = calloc(1, sizeof(ip_info));
+    tnl_status.IpInfo->Ip = strdup(ipaddr_ntoa(&dns_ip4));
+    tnl_status.IpInfo->DNS = strdup(ipaddr_ntoa(&dns_ip4));
+    tnl_status.IpInfo->MTU = 65535;
+
+    if (tnl_status.IpInfo->Subnet) free(tnl_status.IpInfo->Subnet);
+    uint32_t netmask = (0xFFFFFFFFUL << (32 - bits)) & 0xFFFFFFFFUL;
+    netmask = htonl(netmask);
+    tnl_status.IpInfo->Subnet = strdup(ipaddr_ntoa(&netmask));
```

Comment:
`ipaddr_ntoa` takes a pointer to `ip_addr_t`, not a pointer to uint32_t. You can use `IPADDR4_INIT` to create an appropriate ip_addr_t.